lib/puppet/pops/types/p_object_type.rb

```diff
@@ -333,6 +372,86 @@ def initialize(name_or_i12n_hash, i12n_hash_expression = nil)
     end
   end
 
+  # @api private
+  def new_function(loader)
+    @new_function ||= create_new_function(loader)
+  end
+
+  # @api private
+  def create_new_function(loader)
```

Comment:
The documentation of this needs to go into the documentation of the `new` function as that is the only documentation that will be visible to users. At some later point documentation needs to be extracted from object types.